### PR TITLE
MAT-1012 qdm version/packaging bug

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/main/java/mat/model/cql/CQLIncludeLibrary.java [hunk 1]
```diff
@@ -123,11 +126,12 @@ public boolean equals(Object arg0) {
             return false;
         }
 
+        // (cqlIncludeLibrary.libraryModelType == libraryModelType || libraryModelType != null && libraryModelType.equals(cqlIncludeLibrary.libraryModelType))
         if (cqlIncludeLibrary.cqlLibraryId.equals(cqlLibraryId) &&
                 cqlIncludeLibrary.aliasName.equals(aliasName) &&
                 cqlIncludeLibrary.cqlLibraryName.equals(cqlLibraryName) &&
                 cqlIncludeLibrary.version.equals(version) &&
-                cqlIncludeLibrary.libraryModelType.equals(libraryModelType)) {
+                (Objects.equals(libraryModelType, cqlIncludeLibrary.libraryModelType))) {
```
Comment (serhii-ilin) [inline src/main/java/mat/model/cql/CQLIncludeLibrary.java:134]: We had a NullPointer here

File: src/main/java/mat/model/cql/CQLIncludeLibrary.java [hunk 2]
```diff
@@ -25,7 +28,7 @@
 
     private String measureId;
 
-    private String libraryModelType;
+    private String libraryModelType = ModelTypeHelper.QDM;
```
Comment (serhii-ilin) [inline src/main/java/mat/model/cql/CQLIncludeLibrary.java:31]: The library's model type is not persisted in Cator'x model XML. 
We should probably change that.

File: src/main/java/mat/model/cql/CQLIncludeLibrary.java (hunk 1 above)
Comment (serhii-ilin) [reply src/main/java/mat/model/cql/CQLIncludeLibrary.java:134]: libraryModelType was null

**Review by cday-semanticbits — APPROVED** [review]
Nice clean up as well as fixing the issue.

File: src/main/java/mat/server/service/impl/MeasurePackageServiceImpl.java [hunk 3]
```diff
@@ -556,8 +556,10 @@ private void createAndSaveExportsAndArtifacts(MeasureExport export, boolean shou
         measure.setExportedDate(new Date());
         measureDAO.save(measure);
         measureExportDAO.save(export);
-        //New for FHIR we create a CQL_LIBRARY in draft state for the measure so microservices can find it.
-        measureLibraryService.exportDraftCQLLibraryForMeasure(measure);
+        if (measure.isFhirMeasure()) {
```
Comment (serhii-ilin) [inline src/main/java/mat/server/service/impl/MeasurePackageServiceImpl.java:559]: @cday-semanticbits 
JFYI - this custom FHIR code caused issues to QDM packaging so we now run it for FHIR measured only

File: src/main/java/mat/server/CQLUtilityClass.java [hunk 4]
```diff
@@ -351,7 +351,7 @@ public static String getXMLFromCQLModel(CQLModel cqlModel) {
             marshaller.marshal(cqlModel);
             xml = stream.toString();
         } catch (MarshalException | ValidationException | IOException | MappingException e) {
-            e.printStackTrace();
+            logger.error("Error in  getXMLFromCQLModel: " + e.getMessage(), e);
```
Comment (adongare) [inline src/main/java/mat/server/CQLUtilityClass.java:354]: Message will be there in stacktrace. I think no need of `e.getMessage`

**Review by adongare — APPROVED** [review]

File: src/main/java/mat/server/service/impl/MeasurePackageServiceImpl.java (hunk 3 above)
Comment (cday-semanticbits) [reply src/main/java/mat/server/service/impl/MeasurePackageServiceImpl.java:559]: Good catch, should have been for FHIR only.

**Pull request merged**